### PR TITLE
1.0.0-b28

### PR DESCRIPTION
**Pull request by vinniefalco**
* Split out and rename test stream classes
* Restyle async result constructions
* Fix HTTP split parse edge case

### CODE REVIEW

File: include/beast/http/impl/basic_parser_v1.ipp [hunk 1]
```diff
@@ -94,7 +94,9 @@ operator=(basic_parser_v1<
     http_major_ = other.http_major_;
     http_minor_ = other.http_minor_;
     status_code_ = other.status_code_;
+    flags_ = other.flags_;
```
Comment (wilsonianb) [inline include/beast/http/impl/basic_parser_v1.ipp]: This duplicates line 91

File: include/beast/http/basic_parser_v1.hpp [hunk 2]
```diff
@@ -280,12 +281,12 @@ class basic_parser_v1 : public detail::parser_base
     std::uint64_t content_length_;
     pmf_t cb_;
     state s_              : 8;
-    unsigned flags_       : 8;
     unsigned fs_          : 8;
     unsigned pos_         : 8; // position in field state
     unsigned http_major_  : 16;
     unsigned http_minor_  : 16;
     unsigned status_code_ : 16;
+    unsigned flags_ = 0;
```
Comment (seelabs) [inline include/beast/http/basic_parser_v1.hpp]: Probably want to declare this `uint16_t`. We don't need the whole width of an int.

File: include/beast/http/impl/parse.ipp [hunk 3]
```diff
@@ -41,7 +41,6 @@ class parse_op
             , db(sb_)
             , p(p_)
         {
-            BOOST_ASSERT(! p.complete());
```
Comment (seelabs) [inline include/beast/http/impl/parse.ipp]: Given how `paused` is now used, this assert is correct. We can add it back.

Comment (vinniefalco) [inline include/beast/http/impl/parse.ipp]: It was going off but I guess that's from being uninitialized...

File: extras/beast/test/string_ostream.hpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
+//
+// Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
```
Comment (vinniefalco) [inline extras/beast/test/string_ostream.hpp]: willfix

File: include/beast/http/impl/basic_parser_v1.ipp (hunk 1 above)
Comment (vinniefalco) [inline include/beast/http/impl/basic_parser_v1.ipp]: willfix

File: extras/beast/test/string_istream.hpp [hunk 5]
```diff
@@ -124,4 +130,4 @@ class string_stream
 } // test
 } // beast
 
-#endif
+#endif
```
Comment (vinniefalco) [inline extras/beast/test/string_istream.hpp]: willfix

**Comment by vinniefalco** [comment]
@wilsonianb Added a commit for documentation, preview here:
http://vinniefalco.github.io/stage/beast/

File: include/beast/websocket/stream.hpp [hunk 6]
```diff
@@ -1118,12 +1118,17 @@ class stream : public detail::stream_base
...
-        eventually return @ref error::closed.
+        During reads, the implementation handles control frames as
+        follows:
+
+        @li A pong frame is sent when a ping frame is received.
+
+        @li The @ref ping_callback is invoked when a ping frame
+            or pong frame is received.
+
+        @li The WebSocket close procedure is start if a close frame
```
Comment (wilsonianb) [inline include/beast/websocket/stream.hpp]: `start` --> `started`
Ditto below
